src/cacheout/cache.py

```diff
@@ -381,17 +392,23 @@ def _set(self, key: t.Hashable, value: t.Any, ttl: t.Optional[T_TTL] = None) ->
         if ttl is None:
             ttl = self.ttl
 
+        old_value = UNSET
         if key not in self._cache:
             self.evict()
+        else:
+            old_value = self._cache[key]
 
         # Delete key before setting it so that it moves to the end of the OrderedDict key list.
         # Needed for cache strategies that rely on the ordering of when keys were last inserted.
-        self._delete(key, RemovalCause.SET)
+        self._delete(key, RemovalCause._IGNORE)
```

Comment:
Would rather make the cause argument `cause: t.Optional[RemovalCause] = None` instead of adding `RemovalCause._IGNORE` to the enum (it's not a valid value).